### ISSUE TITLE
Attribute resolution of an Enum member that is an instance of a subclass of the mixin class does not follow MRO

### ISSUE THREAD

**Issue by blhsing** [python/cpython#116487]
# Bug report

### Bug description:

As reported in https://stackoverflow.com/questions/78120133/python-enum-of-classes-with-polymorphism .

Since the idea of a mixin class to an `Enum` is to make the members behave like instances of the mixin class, I think it's reasonable to expect that attribute resolution of a member whose value is an instance of a subclass of the mixin class follows MRO as well. Currently the attribute resolution appears to stick to the base mixin class.

Minimal reproducible example:
```python
from enum import Enum

class Base:
    name = 'base'

class Child(Base):
    name = 'child'

class BaseEnum(Base, Enum):
    FOO = Child()

print(BaseEnum.FOO.name) # outputs 'base' when 'child' is expected
print(BaseEnum.FOO.value.name) # outputs 'child' as expected
```

### CPython versions tested on:

3.12

### Operating systems tested on:

Linux, Windows

**Comment by ethanfurman** [comment]
```python
>>> isinstance(BaseEnum.FOO, Base)
True

>>> isinstance(BaseEnum.FOO, Child)
False
```

`Child` is not in `BaseEnum`s mro.

What is the real world use-case?

**Comment by blhsing** [comment]
> ```python
> >>> isinstance(BaseEnum.FOO, Base)
> True
> 
> >>> isinstance(BaseEnum.FOO, Child)
> False
> ```
> 
> `Child` is not in `BaseEnum`s mro.

Right. It isn't clear why the `Activations` class in the SO question needs to be an `Enum` at all in the first place, and I can't think of a good real-world use case myself. I simply concur with the OP of the SO question that the behavior is somewhat unexpected/counter-intuitive. That's all.
